eos-updater/com.endlessm.Updater.xml

```diff
@@ -2,49 +2,297 @@
 '-//freedesktop//DTD D-BUS Object Introspection 1.0//EN'
 'http://www.freedesktop.org/standards/dbus/1.0/introspect.dtd'>
 <node>
+  <!--
+    com.endlessm.Updater:
+    @short_description: Endless OS updater control interface
+
+    Interface to query the state of the OS updater, and to control it: checking
+    for, downloading and applying updates.
+
+    The updater moves through a series of states (see the `State` property and
+    `StateChanged` signal), with state transitions triggered by the appropriate
+    method call, or by an error in an operation. It will return a
+    `com.endlessm.Updater.Error.WrongState` error if you attempt to perform an
+    invalid state transition.
```

Comment:
Not one for this PR but I do sometimes wonder whether `Poll` in state `Polling`, `Fetch` in state `Fetching`, etc. shouldn't be no-ops rather than errors.

Reply:
Returning an error gives the caller a bit more information. They should be handling errors discriminately in any case, so should be able to distinguish it from other, more error-y, errors.